local require = require(script.Parent.loader :: ModuleScript).load(script)
local BinderProvider = require("BinderProvider")
local Binder = require("Binder")

return BinderProvider.new(function(self, serviceBag)
	self:Add(Binder.new("Player", require("Player"), self, serviceBag))
	self:Add(Binder.new("Slime", require("Slime"), self, serviceBag))
	self:Add(Binder.new("Character", require("Character"), self, serviceBag))
	self:Add(Binder.new("Overhead", require("Overhead"), self, serviceBag))
	self:Add(Binder.new("Chat", require("Chat"), self))
	self:Add(Binder.new("Leaderboard", require("Leaderboard"), self))
	self:Add(Binder.new("SlimeView", require("SlimeView"), self, serviceBag))
	self:Add(Binder.new("SlimekeeperView", require("SlimekeeperView"), self, serviceBag))
	self:Add(Binder.new("WorldLeaderboard", require("WorldLeaderboard"), self))
end)
